website/static/js/fangorn.js

```diff
@@ -2541,8 +2543,14 @@ function _dropLogic(event, items, folder) {
             });
         } else {
             toMove.ready.forEach(function(item) {
-                doItemOp.call(tb, copyMode === 'move' ? OPERATIONS.MOVE : OPERATIONS.COPY, folder, item, undefined, 'replace');
             });
+            var moveInterval = setInterval(function() {
+                if(toMove.ready.length > 0) {
+                    doItemOp.call(tb, copyMode === 'move' ? OPERATIONS.MOVE : OPERATIONS.COPY, folder, toMove.ready.pop(), undefined, 'replace');
+                } else {
+                    clearInterval(moveInterval);
+                }
+            }, MILLISECONDS_PER_MOVE_REQUEST);
```

Comment:
Seems like a good solution - using `setInterval` which takes a function and the milliseconds in between evaluating that function.  You're going through all the files and popping them off, when there are no more files you `clearInterval` to stop executing that function.

Reply:
I'm far from an expert with JS, but it's my understanding that using `setInterval` when avoidable is usually considered bad practice.

We have, in `osfHelpers.js`, a `throttle` function stolen from `underscore.js` specifically for cases like this.

Reply:
Good point. I hadn't seen that before, but [according to this comment from stack overflow](https://stackoverflow.com/a/731625/3579517), `setInterval()` is intended for extra precision at the cost of using up more resources, which doesn't fit our use case.